### PR TITLE
[ticket/17100] Introduce twig macros for commonly used form elements

### PR DESCRIPTION
**Pull request by marc1706**
See the ticket on thoughts behind these changes and why it certainly does not cover all of the HTML in PHP files.

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-17100

### CODE REVIEW

**Comment by hanakin** [comment]
Of note we opted for functions for the icons to avoid all the imports. We might want to consider the same here as there will be quite a lot of them

**Comment by marc1706** [comment]
@hanakin updated the PR in that way

File: phpBB/adm/index.php [hunk 1]
```diff
@@ -53,12 +53,19 @@
...
 
 // Set custom style for admin area
-$template->set_custom_style(array(
-	array(
-		'name' 		=> 'adm',
-		'ext_path' 	=> 'adm/style/',
-	),
-), $phpbb_admin_path . 'style');
+/** @var \phpbb\template\base $template */
+$template->set_custom_style(
```
Comment (marc1706) [inline phpBB/adm/index.php:57]: This was changed to support a single form macro file in both adm and prosilver ... so might not be needed

File: phpBB/phpbb/template/twig/environment.php [hunk 2]
```diff
@@ -75,7 +75,7 @@ public function __construct(\phpbb\config\config $phpbb_config, \phpbb\filesyste
 
 		$options = array_merge(array(
 			'cache'			=> (defined('IN_INSTALL')) ? false : $cache_path,
-			'debug'			=> false,
+			'debug'			=> true,
```
Comment (CHItA) [inline phpBB/phpbb/template/twig/environment.php:78]: Why is this changed?

**Review by CHItA — COMMENTED** [review]
Otherwise looks good to me.

File: phpBB/phpbb/template/twig/environment.php (hunk 2 above)
Comment (marc1706) [reply phpBB/phpbb/template/twig/environment.php:78]: I think I changed this for some more debugging and then forgot to change it back. Will revert this line.

File: phpBB/phpbb/template/twig/twig.php [hunk 3]
```diff
@@ -90,6 +90,7 @@ public function __construct(
 		{
 			$this->twig->addExtension($extension);
 		}
+		$this->twig->addExtension(new \Twig\Extension\DebugExtension());
```
Comment (rubencm) [inline phpBB/phpbb/template/twig/twig.php:93]: This line makes the development environment to fail

File: phpBB/includes/acp/acp_board.php [hunk 4]
```diff
@@ -28,16 +33,32 @@ class acp_board
...
+
 	function main($id, $mode)
 	{
 		global $user, $template, $request, $language;
 		global $config, $phpbb_root_path, $phpEx;
 		global $cache, $phpbb_container, $phpbb_dispatcher, $phpbb_log;
 
-		/** @var \phpbb\language\language $language Language object */
-		$language = $phpbb_container->get('language');
+		if (!$language)
```
Comment (rubencm) [inline phpBB/includes/acp/acp_board.php:51]: I think this condition can never happen

File: phpBB/includes/acp/acp_board.php [hunk 5]
```diff
@@ -1019,11 +1071,15 @@ function quick_reply($value, $key)
 	*/
 	function timezone_select($value, $key)
 	{
-		global $template, $user;
+		global $user;
```
Comment (rubencm) [inline phpBB/includes/acp/acp_board.php:1074]: there is already a property for user in this class

File: phpBB/includes/functions_acp.php [hunk 6]
```diff
@@ -294,34 +320,125 @@ function build_cfg_template($tpl_type, $key, &$new_ary, $config_key, $vars)
 			$min = isset($tpl_type[1]) ? (int) $tpl_type[1] : false;
 			$max = isset($tpl_type[2]) ? (int) $tpl_type[2] : false;
 
-			$tpl = '<input id="' . $key . '" type="' . $tpl_type[0] . '"' . (( $min !== false ) ? ' min="' . $min . '"' : '') . (( $max !== false ) ? ' max="' . $max . '"' : '') . ' name="' . $name . '" value="' . $new_ary[$config_key] . '" />';
+			$tpl = [
+				'tag'		=> 'input',
+				'id'		=> $key,
+				'type'		=> $tpl_type[0],
+				'name'		=> $name,
+				'min'		=> $min !== false ? $min : '',
```
Comment (rubencm) [inline phpBB/includes/functions_acp.php:328]: ```$min ?: ''```

shorter version, same for max

File: phpBB/includes/functions_acp.php [hunk 7]
```diff
@@ -294,34 +320,125 @@ function build_cfg_template($tpl_type, $key, &$new_ary, $config_key, $vars)
...
-			$cols = (int) $tpl_type[2];
-
-			$tpl = '<textarea id="' . $key . '" name="' . $name . '" rows="' . $rows . '" cols="' . $cols . '">' . $new_ary[$config_key] . '</textarea>';
+			$tpl = [
+				'tag'		=> 'textarea',
+				'id'		=> $key,
+				'name'		=> $name,
+				'rows'		=> (int) $tpl_type[1],
+				'cols'		=> (int) $tpl_type[2],
+				'content'	=> $new_ary[$config_key],
```
Comment (rubencm) [inline phpBB/includes/functions_acp.php:365]: In the template this is uppercase, i am not sure if this matter, but sometimes you pass lowercase variables and sometime uppercase to the template

File: phpBB/includes/functions_acp.php [hunk 8]
```diff
@@ -294,34 +320,125 @@ function build_cfg_template($tpl_type, $key, &$new_ary, $config_key, $vars)
...
+			];
 		break;
 
 		case 'radio':
-			$key_yes	= ($new_ary[$config_key]) ? ' checked="checked"' : '';
-			$key_no		= (!$new_ary[$config_key]) ? ' checked="checked"' : '';
-
 			$tpl_type_cond = explode('_', $tpl_type[1]);
-			$type_no = ($tpl_type_cond[0] == 'disabled' || $tpl_type_cond[0] == 'enabled') ? false : true;
+			$type_no = !(($tpl_type_cond[0] == 'disabled' || $tpl_type_cond[0] == 'enabled'));
```
Comment (rubencm) [inline phpBB/includes/functions_acp.php:371]: This don't matter much, but parenthesis here is redundant

**Comment by rubencm** [comment]
What I did in acp_storage would look much better with all of this :sweat_smile:

**Review by rubencm — APPROVED** [review]

File: phpBB/includes/functions_acp.php (hunk 6 above)
Comment (marc1706) [reply phpBB/includes/functions_acp.php:328]: The check is actually stricter than what you proposed and fully mirrors the previous check.

File: phpBB/includes/functions_acp.php (hunk 7 above)
Comment (marc1706) [reply phpBB/includes/functions_acp.php:365]: I've actually opted for this for a better distinction between data supplied to the forms twig function and actual template data. The uppercase version is created in the twig forms extension where the input data (form_data) gets translated to the data that gets output to the template.

File: phpBB/includes/functions_acp.php (hunk 8 above)
Comment (marc1706) [reply phpBB/includes/functions_acp.php]: Cleaned it up even a tiny bit better ;)

**Review by CHItA — APPROVED** [review]
